http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpRequest.java

```diff
@@ -116,8 +116,49 @@
  * @since 1.0
  */
 @Internal
-public class NettyHttpRequest<T> extends AbstractNettyHttpRequest<T> implements HttpRequest<T>, PushCapableHttpRequest<T>, io.micronaut.http.FullHttpRequest<T>, ServerHttpRequest<T> {
+public final class NettyHttpRequest<T> extends AbstractNettyHttpRequest<T> implements HttpRequest<T>, PushCapableHttpRequest<T>, io.micronaut.http.FullHttpRequest<T>, ServerHttpRequest<T> {
     private static final Logger LOG = LoggerFactory.getLogger(NettyHttpRequest.class);
+    private static final BiConsumer<String, Object> CLEANUP_ATTRIBUTES = new BiConsumer<>() {
```

Comment:
I think it's possible to do this without a variable. put the lambda body in a static method and make it a method reference, then i believe it's cached.